qualtrix/api.py

```diff
@@ -46,15 +52,53 @@ async def get_response(request: ResponseModel):
 
 
 @router.post("/redirect")
-async def get_redirect(request: RedirectModel):
+async def intake_redirect(request: RedirectModel):
+    start_time = time.time()
     try:
-        email = client.get_email(request.surveyId, request.responseId)
-        contact = client.get_contact(settings.DIRECTORY_ID, email)
-        distribution = client.get_distribution(settings.DIRECTORY_ID, contact["id"])
-        return client.get_link(request.targetSurveyId, distribution["distributionId"])
+        directory_entry = client.create_directory_entry(
+            request.email,
+            request.firstName,
+            request.lastName,
+            settings.DIRECTORY_ID,
+            settings.MAILING_LIST_ID,
+        )
+
+        email_distribution = client.create_email_distribution(
+            directory_entry["contactLookupId"],
+            settings.LIBRARY_ID,
+            settings.INVITE_MESSAGE_ID,
+            settings.MAILING_LIST_ID,
+            request.targetSurveyId,
+        )
+        link = client.get_link(request.targetSurveyId, email_distribution["id"])
+
+        # If link creation succeeds, create reminders while the link is returned
+        create_task(create_reminder_distributions(email_distribution["id"]))
+        create_task(add_user_to_contact_list(link["link"], directory_entry["id"]))
```

Comment:
We can fire and forget both these tasks. They are side effects of the Rules Survey completion and are unrelated to the redirect flow (barring consuming some of its return values)

Putting these two calls in another endpoint will be the primary target for redirect 2.1